components/FeatureHighlight/src/private/MDCFeatureHighlightView+Private.m

```diff
@@ -178,12 +180,70 @@ - (void)setOuterHighlightColor:(UIColor *)outerHighlightColor {
   _outerLayer.fillColor = _outerHighlightColor.CGColor;
 }
 
+- (void)setTitleFont:(UIFont *)titleFont {
+  _titleFont = titleFont;
+
+  [self updateTitleFont];
+}
+
+- (void)updateTitleFont {
+  // If we have a custom font apply it to the label.
+  // If not, fall back to the Material specified font.
+  if (_titleFont) {
+    // If we are automatically adjusting for Dynamic Type resize the font based on the text style
+    if (_mdc_adjustsFontForContentSizeCategory) {
+      _titleLabel.font =
+          [_titleFont mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
+                                   scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+    } else {
+      _titleLabel.font = _titleFont;
+    }
+  } else {
+    if (_mdc_adjustsFontForContentSizeCategory) {
+      _titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:kTitleTextStyle];
+    } else {
+      _titleLabel.font = [UIFont mdc_standardFontForMaterialTextStyle:kTitleTextStyle];
+    }
+  }
+
+  [self setNeedsLayout];
+}
+
 - (void)setTitleColor:(UIColor *)titleColor {
   _titleColor = titleColor;
 
   _titleLabel.textColor = titleColor;
 }
 
+- (void)setBodyFont:(UIFont *)bodyFont {
```

Comment:
I think this should read:
```
- (void)setBodyFont:(UIFont *)bodyFont {
  if (bodyFont == nil) {
    _bodyLabel.font = [Typography bodyTextFont];
  } else {
    _bodyLabel.font = bodyFont;
  }
  
  if (_mdc_adjusts...) {
    _bodyLabel.font =
        [_bodyFont mdc_fontSizedForMaterialTextStyle:kBodyTextStyle
                                                scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
  }
}
```

Reply:
and that changes to the _mdc_adjust bool and sizeCategoryChanges should just call `[self setBodyFont:self.bodyFont]`

Reply:
fontSizedForMaterialTextStyle is intended to scale a custom font.
preferredFontForStyle is intended to eventually migrate to UIContentSizeAdjusting, which is Apple's prescribed API.

Per the Design Doc, properties (set*Font and mdc_setAdjusting) and the sizeCategoryChange handler (and any other font changing method) should call update*Font.

Reply:
Sorry i had a minor error in my suggestion. I corrected it (inside the if (_mdc_adjusts...){ clause i changed the object calling of the mdc_fontSizedForMaterial.. method)

If we want the updateFont to be in a different method fine but it should source from the label.font not the property that we have created.

Reply:
Unfortunately, it won't be orthogonal.

setFont - Comic@22
enable Dynamic Type auto-adjusting : Comic@22 gets scaled to Comic@26
disable Dynamic Type auto-adjusting :
We need the original Comic@22, but we can't get back to there from Comic@26

We need to store the customFont as explicit state.

Reply:
Yes we talked about that and we decided that it was acceptable to lose size. We would use the "large" or what apple considers default.

Reply:
I expect this unit test to pass:

Typogragrphy.fontloader = ComicSans
featureHighlight.mdc_adjust = true
XCTAssert(featureHighlight.font.face, comicSans)
featureHighlight.mdc_adjust = false
XCTAssert(featureHighlight.font.face, comicSans)

With the current logic both fail.